src/main/java/de/tum/in/www1/artemis/web/rest/FileUploadSubmissionResource.java

```diff
@@ -246,50 +253,47 @@ public FileUploadSubmissionResource(FileUploadSubmissionRepository fileUploadSub
                                 HeaderUtil.createFailureAlert(applicationName, true, "fileUploadExercise", "exerciseEmpty", "The exercise belonging to the participation is null."))
                         .body(null);
             }
+
+            // make sure sensitive information are not sent to the client
+            fileUploadExercise.filterSensitiveInformation();
         }
         else {
             return ResponseEntity.badRequest().headers(HeaderUtil.createFailureAlert(applicationName, true, "fileUploadExercise", "wrongExerciseType",
                     "The exercise of the participation is not a file upload exercise.")).body(null);
         }
 
-        // users can only see their own submission (to prevent cheating), TAs, instructors and admins
-        // can see all answers
-        if (!authCheckService.isOwnerOfParticipation(participation)
-                && !authCheckService.isAtLeastTeachingAssistantInCourse(fileUploadExercise.getCourse(), userService.getUserWithGroupsAndAuthorities())) {
-            return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
-        }
-
-        // if no results, check if there are really no results or the relation to results was not
-        // updated yet
-        if (participation.getResults().size() <= 0) {
-            List<Result> results = resultRepository.findByParticipationIdOrderByCompletionDateDesc(participation.getId());
-            participation.setResults(new HashSet<>(results));
+        // Students can only see their own models (to prevent cheating). TAs, instructors and admins can see all models.
```

Comment:
```suggestion
        // Students can only see their own file uploads (to prevent cheating). TAs, instructors and admins can see all file uploads.
```